src/worker.cpp

```diff
@@ -1101,13 +1103,100 @@ void Worker::populateDbus(const types::VPDMapVariant& parsedVpdMap,
     }
 }
 
+std::string Worker::getAssetTag(const types::VPDMapVariant& i_parsedVpdMap)
+{
+    std::string l_assetTag;
+
+    try
+    {
+        // system VPD will be in IPZ format.
+        if (auto l_parsedVpdMap =
+                std::get_if<types::IPZVpdMap>(&i_parsedVpdMap))
+        {
+            auto l_itrToVsys = (*l_parsedVpdMap).find(constants::recVSYS);
+            if (l_itrToVsys != (*l_parsedVpdMap).end())
+            {
+                std::string l_tmKwdValue;
+                vpdSpecificUtility::getKwVal(l_itrToVsys->second,
+                                             constants::kwdTM, l_tmKwdValue);
+
+                std::string l_seKwdValue;
+                vpdSpecificUtility::getKwVal(l_itrToVsys->second,
+                                             constants::kwdSE, l_seKwdValue);
+
+                l_assetTag = std::string{"Server-"} + l_tmKwdValue +
+                             std::string{"-"} + l_seKwdValue;
+            }
+            else
+            {
+                throw std::runtime_error(
+                    "VSYS record not found in parsed VPD map to extract Asset tag.");
+            }
+        }
+        else
+        {
+            throw std::runtime_error(
+                "Inavlid VPD type recieved to extract Asset tag.");
+        }
+    }
+    catch (const std::exception& l_ex)
+    {
+        // TODO:Log PEL with below description.
+        logging::logMessage("Asset tag can't be formed. Error = " +
+                            std::string(l_ex.what()));
+    }
+
+    return l_assetTag;
+}
+
 void Worker::publishSystemVPD(const types::VPDMapVariant& parsedVpdMap)
 {
     types::ObjectMap objectInterfaceMap;
 
     if (std::get_if<types::IPZVpdMap>(&parsedVpdMap))
     {
         populateDbus(parsedVpdMap, objectInterfaceMap, SYSTEM_VPD_FILE_PATH);
+
+        try
+        {
+            if (m_isFactoryResetDone)
```

Comment:
1. For testing purpose, if we/test team clears inventory cache using rm -rf on /var/lib/phosphor-inventory-manager/xyz...
2. Yes.. lets take an example where the new BMC card has symlink and this folder /var/lib/phosphor-inventory-manager is empty..

Reply:
> For testing purpose, if we/test team clears inventory cache using rm -rf on /var/lib/phosphor-inventory-manager/xyz...

In that case clear the symlink as well I guess.

> Yes.. lets take an example where the new BMC card has symlink and this folder /var/lib/phosphor-inventory-manager is empty/incorrect Asset tag found on BMC cache

a) If symlink is present how the data is empty?
b) In case of incorrect asset tag, Use GUI to update the correct one. The same will be updated on Dbus.

Reply:
a) If symlink is present how the data is empty?

but there is a possibility of data corruption in the newly inserted BMC's cache memory.. where /var/lib/vpd/vpd_inventory.json is present &
/var/lib/phosphor-inventory-manager/ is not present.

b) ok.

Reply:
> but there is a possibility of data corruption

If we go by these type of corruption then we can have multiple scenarios We can't handle all of them. In these cases, BMC should be cleared before using in another system,
Also, I am not sure if we can pull a BMC from another system and put it directly into another system.

Reply:
As discussed in the channel, BMC should be in manufacturing reset state before plugging it into any system. Hence closing this conversation.